sdk/core/azure-core-serializer-json-gson/src/main/java/com/azure/core/serializer/json/gson/GsonJsonSerializer.java

```diff
@@ -42,16 +41,39 @@
         this.gson = gson;
     }
 
+    @Override
+    public <T> T deserialize(byte[] data, TypeReference<T> typeReference) {
+        if (data == null) {
+            return null;
+        }
+
+        return gson.fromJson(new String(data, UTF_8), typeReference.getJavaType());
```

Comment:
Can this just use the default implementation of calling the InputStream overload?

Reply:
Will change this back to using the default implementation